app/models/operatingsystems/rhcos.rb

```diff
@@ -0,0 +1,43 @@
+class Rhcos < Operatingsystem
+  #
+  # Example PXE URLs:
+  # http://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.5/4.5.6/rhcos-installer-kernel-x86_64
+  # http://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.5/4.5.6/rhcos-installer-initramfs.x86_64.img
+  #
+  PXEFILES = {
+    kernel: 'rhcos-installer-kernel-$arch',
```

Comment:
do i have to set the arch in a special way?

i was able to create a rhcos vm in foreman but it hangs at finding the kernel:
![image](https://user-images.githubusercontent.com/10408311/95346875-92851080-08bc-11eb-8a6d-0951d640b8f2.png)

my OS does look like this:
![image](https://user-images.githubusercontent.com/10408311/95347114-d37d2500-08bc-11eb-82b9-290429b76239.png)

or is it possible that the var here is not working correctly?

Reply:
It's a bug! I am rebasing this PR to fix this, good catch. Here is the diff:

```diff
diff --git a/app/models/operatingsystems/fcos.rb b/app/models/operatingsystems/fcos.rb
index 1d894c54d..8286f1b34 100644
--- a/app/models/operatingsystems/fcos.rb
+++ b/app/models/operatingsystems/fcos.rb
@@ -13,6 +13,10 @@ class Fcos < Operatingsystem
     'fcos'
   end
 
+  def bootfile(medium_provider, type)
+    medium_provider.interpolate_vars(super).to_s
+  end
+
   def pxedir(medium_provider = nil)
     medium_provider.interpolate_vars('prod/streams/$release/builds/$major.$minor/$arch').to_s
   end
diff --git a/app/models/operatingsystems/rhcos.rb b/app/models/operatingsystems/rhcos.rb
index 7a2ceefaf..7fa7c7aab 100644
--- a/app/models/operatingsystems/rhcos.rb
+++ b/app/models/operatingsystems/rhcos.rb
@@ -13,6 +13,10 @@ class Rhcos < Operatingsystem
     'rhcos'
   end
 
+  def bootfile(medium_provider, type)
+    medium_provider.interpolate_vars(super).to_s
+  end
+
   def pxedir(medium_provider = nil)
     medium_provider.interpolate_vars('pub/openshift-v$major/$arch/dependencies/rhcos/$major.$minor/$major.$minor.$release').to_s
   end
```

Thanks for testing!